### PR TITLE
#323 Migration to JUnit 5

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #323 

Added two puzzles:
* migrate parametrized `MetricsTest` test
* continue to migrate test in other packages

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#378](https://github.com/yegor256/jpeek/issues/378) is assigned to @fabriciofx[/z](https://www.0crat.com/u/fabriciofx), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/56ffc21a-25f2-4236-b70d-b15b4d69a268); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by HDouss** [comment]
@fabriciofx Please note that code coverage is [returned to normal](https://codecov.io/gh/yegor256/jpeek/pull/378/commits) after enabling junit5 and junit4 tests.

**Comment by paulodamaso** [comment]
@0crat assign me

File: src/test/java/org/jpeek/MainTest.java [hunk 1]
```diff
@@ -57,18 +57,30 @@ public void createsXmlReports() throws IOException {
         ).affirm();
     }
 
-    @Test(expected = ParameterException.class)
+    @Test
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/MainTest.java]: @HDouss You can remove the throws clause from the test now

File: src/test/java/org/jpeek/MainTest.java [hunk 2]
```diff
@@ -57,18 +57,30 @@ public void createsXmlReports() throws IOException {
...
+            "Must throw an exception if parameter is invalid",
+            () -> {
+                Main.main("hello");
+                return null;
+            }, new Throws<>(ParameterException.class)
+        ).affirm();
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/MainTest.java]: @HDouss You can remove the throws clause from the test now

File: src/test/java/org/jpeek/MainTest.java [hunk 3]
```diff
@@ -57,18 +57,30 @@ public void createsXmlReports() throws IOException {
...
 
-    @Test(expected = ParameterException.class)
+    @Test
     public void crashesIfInvalidInput() throws IOException {
-        Main.main("hello");
+        new Assertion<>(
+            "Must throw an exception if parameter is invalid",
+            () -> {
+                Main.main("hello");
+                return null;
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/MainTest.java]: @HDouss Avoid using null

File: src/test/java/org/jpeek/MainTest.java [hunk 4]
```diff
@@ -57,18 +57,30 @@ public void createsXmlReports() throws IOException {
...
-            "--target", target.toString()
-        );
+        new Assertion<>(
+            "Must throw an exception if target exists and no overwrite",
+            () -> {
+                Main.main(
+                    "--sources", Paths.get(".").toString(),
+                    "--target", target.toString()
+                );
+                return null;
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/MainTest.java]: @HDouss Here too, please not use null

File: src/test/java/org/jpeek/MainTest.java (hunk 2 above)
Comment (HDouss) [reply src/test/java/org/jpeek/MainTest.java]: @paulodamaso it is not possible here, as `Files.createTempDirectory` throws `IOException`.

**Comment by HDouss** [comment]
@paulodamaso Thanks for your review. Done!

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/jpeek/issues/378#issuecomment-587228378))

@paulodamaso This pull request [#378](https://github.com/yegor256/jpeek/issues/378) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/e9ea5b93-8010-4e38-9da6-7d88c5e8a444); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by HDouss** [comment]
@paulodamaso Can we merge this please?

**Comment by paulodamaso** [comment]
@rultor merge

File: src/test/java/org/jpeek/package-info.java [hunk 5]
```diff
@@ -28,5 +28,9 @@
  * @author Yegor Bugayenko (yegor256@gmail.com)
  * @version $Id$
  * @since 0.1
+ * @todo #323:30min Migrate the tests in the sub-packages: org.jpeek.metrics,
+ * org.jpeek.skeleton and org.jpeek.web to junit 5, so they won't
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/package-info.java]: @HDouss This puzzle is not formatted correctly: every line after the first must have one space indentation

**Comment by HDouss** [comment]
@paulodamaso Thanks! Solved!

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by HDouss** [comment]
@paulodamaso Sorry, solved again...

**Comment by HDouss** [comment]
@paulodamaso This time it was the length of puzzle wording. It would be nice that qulice could report such problems

**Comment by HDouss** [comment]
@paulodamaso Could you please try to remerge?

**Comment by paulodamaso** [comment]
@HDouss There is a 0pdd checking tool. I don't know why it isn't failing the CI, though

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 16 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Pull request merged**